### PR TITLE
feat: Add zip in multiple network calls

### PR DESCRIPTION
**Pull request by anhanh11001**
Details:
- Add zip for flowable in multiple network calls for similar events
- Make favorite button works in SimilarEvent list

Fixes: #2009

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 1]
```diff
@@ -494,6 +494,7 @@ class EventDetailsFragment : Fragment() {
             onEventClick = eventClickListener
             onFavFabClick = favFabClickListener
         }
+
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: Extra file committed, reset looks good to me.

**Comment by nikit19** [comment]
PR looks good. May be try re running the build

**Comment by anhanh11001** [comment]
@nikit19 updated. I just didn't know why build failed earlier. I basically changed nothing

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt (hunk 1 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: @anhanh11001

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: Still there

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: fixed

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: fixed

**Pull request merged**